src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -205,6 +328,46 @@ IN_PROCESS_APPLICATION::~IN_PROCESS_APPLICATION()
     Recycle();
 }
 
+REQUEST_NOTIFICATION_STATUS
+IN_PROCESS_APPLICATION::OnAsyncCompletion(
+    IHttpContext*           pHttpContext,
+    DWORD                   cbCompletion,
+    HRESULT                 hrCompletionStatus
+)
+{
+    HRESULT hr;
+    IN_PROCESS_STORED_CONTEXT* pInProcessStoredContext = NULL;
+
+    hr = GetInProcessStoredContext(pHttpContext, &pInProcessStoredContext);
+    if (FAILED(hr))
+    {
```

Comment:
We need set http 500 error status before calling finish request

Reply:
this is done, don't know why the diff is still there.